MatrixSDKExtensions/VoIP/Jingle/MXJingleCallStackCall.m

```diff
@@ -602,7 +599,34 @@ - (void)setCameraPosition:(AVCaptureDevicePosition)theCameraPosition
 
 #pragma mark - Private methods
 
-//  Not used for now, will be in future
+- (void)checkTheCallIsOnHold
+{
+    NSArray<RTC_OBJC_TYPE(RTCRtpTransceiver) *> *activeReceivers = [self->peerConnection.transceivers filteredArrayUsingPredicate:[NSPredicate predicateWithBlock:^BOOL(RTC_OBJC_TYPE(RTCRtpTransceiver) *transceiver, NSDictionary<NSString *,id> * _Nullable bindings) {
+        
+        RTCRtpTransceiverDirection direction = RTCRtpTransceiverDirectionStopped;
+        if ([transceiver currentDirection:&direction])
+        {
+            if (direction == RTCRtpTransceiverDirectionInactive ||
+                direction == RTCRtpTransceiverDirectionSendOnly ||
+                direction == RTCRtpTransceiverDirectionStopped)
+            {
+                return NO;
+            }
+        }
+        
+        return YES;
+    }]];
+    
+    //  if there is no active receivers left, we can say this call is holded
+    if (activeReceivers.count == 0)
+    {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [self.delegate callStackCallDidHold:self];
```

Comment:
Should there be a call to say it's un-held here if there is an active receiver? Is this relying on AddStream being called when the call is un-held?

Reply:
Yes, it's relying on `didAddStream` call.

Reply:
ok - as long as this is reliable

Reply:
I didn't like this afterwards, as we can add more streams to calls in future, it may lead to unexpected results. Code updated.